server/visualizer/server.go

```diff
@@ -0,0 +1,49 @@
+package visualizer
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/labstack/echo/v4"
+)
+
+func Echo(g *echo.Group, c Config) error {
+	s, err := NewServices(c)
+	if err != nil {
+		return err
+	}
+
+	initEcho(g, c, s)
+	return nil
+}
+
+func initEcho(g *echo.Group, c Config, s Services) {
+	h, err := NewHandler(s.CMS, c.DataModelKey, c.TemplateModelKey)
+	if err != nil {
+		panic("failed to init echo")
+	}
+	g.GET("/viz/:pid", h.fetchRoot(s.CMS), authMiddleware(c.VToken))
```

Comment:
```suggestion
	g.GET("/viz/:pid", h.fetchRoot())
```

This API should be public